components/expenses/ExpenseFormItems.js

```diff
@@ -119,8 +120,12 @@ class ExpenseFormItems extends React.PureComponent {
     }
   }
 
-  onCurrencyChange = newCurrency => {
-    this.props.form.setFieldValue('currency', newCurrency);
+  onCurrencyChange = async newCurrency => {
+    /* If we are calling setFieldValue in response to a field change and has validations
+     * we should set the field to touched; https://github.com/jaredpalmer/formik/issues/2059
+     */
+    await this.props.form.setFieldValue('currency', newCurrency);
+    this.props.form.setFieldTouched('currency', true);
```

Comment:
Would passing `true` as the 3rd param (`shouldValidate`) have the same effect?

```suggestion
  onCurrencyChange = newCurrency => {
    this.props.form.setFieldValue('currency', newCurrency, true);
```

Reply:
Unfortunately this doesn't work. If we do this we have the same bug where the `touched` is not updated in the first click. 

![bug-select](https://user-images.githubusercontent.com/12435965/160183243-fdad2081-5ada-49cd-b61d-362c611956b9.gif)